File: 2530 Maximal Score After Applying K Operations/2530.cpp
class Solution {
public:
    long long maxKelements(vector<int>& nums, int k) {
        priority_queue<int> pq;
        
        for (int num: nums){
            pq.push(num);
        }
        
        long long ans = 0;

        while (k--){
            int cur = pq.top(); pq.pop();
            ans += cur;
            pq.push((cur+2)/3);
        }

        return ans;
    }
};

File: 2530 Maximal Score After Applying K Operations/2530.py
class Solution:
    def maxKelements(self, nums: List[int], k: int) -> int:
        pq = []
        
        for num in nums:
            heappush(pq, -num)

        ans = 0

        for _ in range(k):
            cur = -heappop(pq)
            ans += cur
            heappush(pq, -ceil(cur/3))

        return ans
                
